### ISSUE TITLE
IMX477: failed to register i2c client during kernel boot

### ISSUE THREAD

**Issue by leigh-johnson** [bitsy-ai/printnanny-os#311]
## **Describe the bug**

I see a report indicating kernel fails to register IMX477 (RPI HD Camera) sensor during first boot. Logs:

```
May 19 17:03:44 [REDACTED:$host] kernel: i2c i2c-10: Failed to register i2c client imx477 at 0x1a (-16)
May 19 17:03:44 [REDACTED:$host] kernel: vc4-drm gpu: bound fef00700.hdmi (ops vc4_hdmi_ops)
May 19 17:03:44 [REDACTED:$host] kernel: i2c i2c-10: of_i2c: Failure registering /soc/i2c0mux/i2c@1/imx477@1a
May 19 17:03:44 [REDACTED:$host] kernel: Registered IR keymap rc-cec
May 19 17:03:44 [REDACTED:$host] kernel: rc rc1: vc4-hdmi-1 as /devices/platform/soc/fef05700.hdmi/rc/rc1
May 19 17:03:44 [REDACTED:$host] kernel: i2c i2c-10: Failed to create I2C device for /soc/i2c0mux/i2c@1/imx477@1a
May 19 17:03:44 [REDACTED:$host] kernel: i2c i2c-22: Added multiplexed i2c bus 10
```

Additionally, the report indicates that `IMX219` is shown in the UI if no sensor is detected. We should provide more meaningful feedback, like "failed to detect camera" and link to:

* troubleshooting doc
* list of supported sensors.

## **To Reproduce**

TDB

## **Expected behavior**

First boot using IMX477 correctly detects the sensor, then sets as default. 

## Issue.txt

```
-----------------------
Build Info:  |
-----------------------
IMAGE_NAME = printnanny-release-image-raspberrypi4-64-20230508161625
DISTRO_PRETTY = PrintNanny Linux 0.7.0 (Garnet)
VARIANT_NAME = PrintNanny OS Core
VARIANT_ID = core
DISTRO = printnanny
DISTRO_NAME = PrintNanny Linux
DISTRO_VERSION = 0.7.0
DISTRO_CODENAME = Garnet
ID_LIKE = BitsyLinux
HOME_URL = https://printnanny.ai
BUG_REPORT_URL = https://github.com/bitsy-ai/printnanny-os/issues
YOCTO_VERSION = 4.1.2
YOCTO_CODENAME = Langdale
MACHINE = raspberrypi4-64
TUNE_PKGARCH = cortexa72
-----------------------
Layer Revisions:      |
-----------------------
meta              = mickledore:ddb298ce894482467d82699d223c5555baad6591
meta-poky         = mickledore:ddb298ce894482467d82699d223c5555baad6591
meta-yocto-bsp    = mickledore:ddb298ce894482467d82699d223c5555baad6591
meta-oe           = mickledore-polkit-src:4321a5fd17fa2c307ca579ae76b921fb4c560bfa
meta-networking   = mickledore-polkit-src:4321a5fd17fa2c307ca579ae76b921fb4c560bfa
meta-python       = mickledore-polkit-src:4321a5fd17fa2c307ca579ae76b921fb4c560bfa
meta-webserver    = mickledore-polkit-src:4321a5fd17fa2c307ca579ae76b921fb4c560bfa
meta-initramfs    = mickledore-polkit-src:4321a5fd17fa2c307ca579ae76b921fb4c560bfa
meta-bitsy        = fix-eof-klipper-moonraker:e1843f5fe4a7bf59f9ac67f8dce4c55d564efc87
meta-microcontroller = mickledore:a848733698649628d0c3e34ae04ced0ee6ce3aee
meta-neural-network = mickledore:39f06bfa582eedc8621df677bb5cf044e9481309
meta-raspberrypi  = master:bf948e0aa87ee0c10bdf85f99b2632915e6d2039
meta-swupdate     = master:19007b03a1d96eca38e12f3bbd7bb296aea50bab
meta-printnanny   = fix-eof-klipper-moonraker:e1843f5fe4a7bf59f9ac67f8dce4c55d564efc87
meta-multimedia   = mickledore-polkit-src:4321a5fd17fa2c307ca579ae76b921fb4c560bfa
```

**Comment by leigh-johnson** [comment]
rpi4b4g-imx708:~$ cat /boot/config.txt | grep imx

```
# Enable Sony RaspberryPi Camera(imx219)
dtoverlay=imx219
# Enable Sony RaspberryPi Camera(imx708)
dtoverlay=imx708
```

We're not loading the imx477 device tree overlay in the latest 0.7.0 rootfs. Putting together a patch now :eyes:

**Comment by leigh-johnson** [comment]
Hah, looks like this was accidentally commented out, ref: https://github.com/agherzan/meta-raspberrypi/issues/1167

**Comment by leigh-johnson** [comment]
:tada: Verified this is fixed in the v0.7.1 nightly.
